### PR TITLE
fix: show audit logs for forgot password flow

### LINKED ISSUE #15150: coderd/audit: ensure audit logs created for user password reset requests
Relates to https://github.com/coder/coder/issues/14232

The following events should end up in the audit logs:
- User password reset requests
- Failed user password reset attempts
- Successful user password reset attempts

Right now, they do not appear to do so.

### PR DESCRIPTION
**Pull request by DanielleMaywood**
Fixes https://github.com/coder/coder/issues/15150

Audit logs for requesting a password reset, and a user updating their password, now show up in the audit log.

Screenshot 2024-10-22 at 12 48 27

### CODE REVIEW

File: codersdk/audit.go [hunk 1]
```diff
@@ -114,6 +115,8 @@ func (a AuditAction) Friendly() string {
 		return "logged out"
 	case AuditActionRegister:
 		return "registered"
+	case AuditActionRequestOneTimePasscode:
+		return "one time passcode requested"
```
Comment (johnstcn) [inline codersdk/audit.go:119]: nit: what do you think about "password reset"?

**Review by johnstcn — APPROVED** [review]
Nice work! The only potential quibble I have is the naming -- perhaps "password reset requested" would be more clear?

File: codersdk/audit.go (hunk 1 above)
Comment (DanielleMaywood) [reply codersdk/audit.go:119]: Happy to make this change

Comment (DanielleMaywood) [reply codersdk/audit.go]: Have made these changes!

**Review by BrunoQuaresma — APPROVED** [review]
🚢 it

**Review by johnstcn — APPROVED** [review]
:shipit:

**Pull request merged**